subsys/event_manager/event_manager.c

```diff
@@ -115,114 +106,20 @@ static void log_event_init(void)
 {
-	size_t event_cnt = __stop_event_types - __start_event_types;
-	size_t event_idx = event_cnt + (is_start ? 0 : 1);
-	size_t trace_evt_id = profiler_event_ids[event_idx];
-
-	if (!IS_ENABLED(CONFIG_EVENT_MANAGER_TRACE_EVENT_EXECUTION) ||
-	    !is_profiling_enabled(trace_evt_id)) {
-		return;
-	}
+};
 
-	struct log_event_buf buf;
-	ARG_UNUSED(buf);
-
-	profiler_log_start(&buf);
-	profiler_log_add_mem_address(&buf, eh);
-	profiler_log_send(&buf, trace_evt_id);
-}
-
-static void trace_event_submission(const struct event_header *eh)
+void __weak trace_event_submission(const struct event_header *eh,
+				const void *trace_info)
 {
-	if (!IS_ENABLED(CONFIG_EVENT_MANAGER_PROFILER_ENABLED)) {
-		return;
-	}
-
-	const struct event_type *et = eh->type_id;
-	size_t event_idx = et - __start_event_types;
-	size_t trace_evt_id = profiler_event_ids[event_idx];
-
-	if (!et->ev_info ||
-	    !et->ev_info->profile_fn ||
-	    !is_profiling_enabled(trace_evt_id)) {
-		return;
-	}
-
-	struct log_event_buf buf;
-	ARG_UNUSED(buf);
-
-	profiler_log_start(&buf);
-
-	if (IS_ENABLED(CONFIG_EVENT_MANAGER_TRACE_EVENT_EXECUTION)) {
-		profiler_log_add_mem_address(&buf, eh);
-	}
-	if (IS_ENABLED(CONFIG_EVENT_MANAGER_PROFILE_EVENT_DATA)) {
-		et->ev_info->profile_fn(&buf, eh);
-	}
+};
 
-	profiler_log_send(&buf, trace_evt_id);
-}
-
-static void trace_register_execution_tracking_events(void)
-{
-	static const char * const labels[] = {EM_MEM_ADDRESS_LABEL};
-	enum profiler_arg types[] = {MEM_ADDRESS_TYPE};
-	size_t event_cnt = __stop_event_types - __start_event_types;
-	uint16_t profiler_event_id;
-
-	ARG_UNUSED(types);
-	ARG_UNUSED(labels);
-
-	/* Event execution start event after last event. */
-	profiler_event_id = profiler_register_event_type(
-				"event_processing_start",
-				labels, types, 1);
-	profiler_event_ids[event_cnt] = profiler_event_id;
-
-	/* Event execution end event. */
-	profiler_event_id = profiler_register_event_type(
-				"event_processing_end",
-				labels, types, 1);
-	profiler_event_ids[event_cnt + 1] = profiler_event_id;
-}
-
-static void trace_register_events(void)
-{
-	for (const struct event_type *et = __start_event_types;
-	     (et != NULL) && (et != __stop_event_types);
-	     et++) {
-		if (et->ev_info) {
-			size_t event_idx = et - __start_event_types;
-			uint16_t profiler_event_id;
-
-			profiler_event_id = profiler_register_event_type(
-				et->name, et->ev_info->log_arg_labels,
-				et->ev_info->log_arg_types,
-				et->ev_info->log_arg_cnt);
-			profiler_event_ids[event_idx] = profiler_event_id;
-		}
-	}
-
-	if (IS_ENABLED(CONFIG_EVENT_MANAGER_TRACE_EVENT_EXECUTION)) {
-		trace_register_execution_tracking_events();
-	}
-}
-
-static int trace_event_init(void)
+int __weak trace_event_init(void)
 {
-	if (IS_ENABLED(CONFIG_EVENT_MANAGER_PROFILER_ENABLED)) {
-		if (profiler_init()) {
-			LOG_ERR("System profiler: "
-				"initialization problem\n");
-			return -EFAULT;
-		}
-		trace_register_events();
-	}
-
 	return 0;
-}
+};
```

Comment:
This semicolon is not needed. Applies to each definition of those weak functions.